core/src/main/scala/org/apache/spark/api/python/PythonRDD.scala

```diff
@@ -866,11 +866,14 @@ class BytesToString extends org.apache.spark.api.java.function.Function[Array[By
 }
 
 /**
- * Internal class that acts as an `AccumulatorParam` for Python accumulators. Inside, it
+ * Internal class that acts as an `AccumulatorV2` for Python accumulators. Inside, it
  * collects a list of pickled strings that we pass to Python through a socket.
  */
-private class PythonAccumulatorParam(@transient private val serverHost: String, serverPort: Int)
-  extends AccumulatorParam[JList[Array[Byte]]] {
+private[spark] class PythonAccumulatorV2(@transient private val serverHost: String, serverPort: Int)
+  extends AccumulatorV2[JList[Array[Byte]], JList[Array[Byte]]] {
```

Comment:
I gave this a shot locally -- haven't tested it though. Does this make sense or do you see a reason this won't work? 

https://github.com/srowen/spark/commit/f13034bfeacb43d816c78a03e891ea46efa43607